api/watch/funcs.go

```diff
@@ -340,7 +340,10 @@ func agentServiceWatch(params map[string]interface{}) (WatcherFunc, error) {
 func makeQueryOptionsWithContext(p *Plan, stale bool) consulapi.QueryOptions {
 	ctx, cancel := context.WithCancel(context.Background())
 	p.setCancelFunc(cancel)
-	opts := consulapi.QueryOptions{AllowStale: stale}
+
+	opts := p.QueryOptions
+	opts.AllowStale = stale
+
```

Comment:
If my suggestion about only adding "peer, namespace, partition" to the Plan struct sounds good to you, we need to construct the QueryOptions here as well,


```
if p.Peer != "" {
    opt.Peer = p.Peer
}
```